src/pip/_internal/req/req_file.py

```diff
@@ -172,6 +174,10 @@ def process_line(
     opts, _ = parser.parse_args(
         shlex.split(options_str), defaults)  # type: ignore
 
+    if opts.use_pep517 is not None:
```

Comment:
I rebased this for my own needs, and I only needed to change this to `if options and options.use_pep517 ...`